[ADD] payment_acquirers: add configuration steps for Razorpay (#2149)
odoo/documentation: content/applications/finance/payment_acquirers/razorpay.rst

===========
Razorpay
===========

`Razorpay <https://razorpay.com/>`_ is an online payments provider established in India and
covering 100+ payment methods.

.. _payment_acquirers/razorpay/configure_dashboard:

Configuration on Razorpay Dashboard
======================================

#. Log into `Razorpay Dashboard <https://dashboard.razorpay.com/>`_ and go to

>    :menuselection:`Settings --> API Keys`. Generate the new keys and copy the values of the :guilabel:`Key Id` and

Reviewer: Although the Content Guidelines state 100th character-limit, it is actually a 70-character limit. The sentence should fit within the limit (indicated by the vertical grey line in the screenshot).

![70th character](https://user-images.githubusercontent.com/104567387/174621121-68d2cfe1-2514-4682-8ce9-3f61aad57732.png)

   :guilabel:`Secret Key` fields and save them for later.
#. | Go to :menuselection:`Settings --> Webhooks` and enter your Odoo database URL followed by
     `/payment/razorpay/webhook` in the :guilabel:`Webhook URL` text field.
   | For example: `https://yourcompany.odoo.com/payment/razorpay/webhook`.

> #. Fill the :guilabel:`Secret` with a password that you generate and save its value for later.

Reviewer: Did you mean "Fill the :guilabel:`Secret` **field** with a password...? I feel a word is missing in the sentence.

I would say "Fill the Fill the :guilabel:`Secret` **field** with a password of your choice and save it for later." In my opinion, this seems more clear and more concise.

> #. Make sure all the remaining checkboxes are ticked.

Reviewer: Maybe add a screenshot to make it more clear and understandable.

#. Click on **Save** to finalize the configuration.

.. _payment_acquirers/razorpay/configure_odoo:

Configuration on Odoo
=====================

#. :ref:`Navigate to the payment acquirer Razorpay <payment_acquirers/add_new>` and change its
   state to :guilabel:`Enabled`.
#. In the :guilabel:`Credentials` tab, fill the :guilabel:`Public Key`, :guilabel:`Secret Key`, and
   :guilabel:`Webhook Secret` with the values you saved at the step
   :ref:`payment_acquirers/razorpay/configure_dashboard`.
#. Configure the rest of the options to your liking.

>    .. important::

Reviewer: ..important:: has 3 spaces before when it should be flush with the left margin (no spaces)

      - If you configure Odoo to capture amounts manually, make sure to set the **Manual Capture** to
        **manual** on Razorpay. Otherwise, the transaction will be blocked in the authorized state in

>         Odoo.. To do so, go to your Razorpay Dashboard and then to :menuselection:`Settings --> Configuration`.

Reviewer: These 3 lines also go beyond the 70 char limit.

There's a double '.' at the end of Odoo, I believe there should be only one.

===========
Razorpay
===========

`Razorpay <https://razorpay.com/>`_ is an online payments provider established in India and

> covering 100+ payment methods.

Reviewer 2: "more than 100"

.. _payment_acquirers/razorpay/configure_dashboard:

Configuration on Razorpay Dashboard
===================================

#. Log into `Razorpay Dashboard <https://dashboard.razorpay.com/>`_ and go to
   :menuselection:`Settings --> API Keys`. Generate the new keys and copy the values of the
   :guilabel:`Key Id` and :guilabel:`Secret Key` fields and save them for later.

> #. | Go to :menuselection:`Settings --> Webhooks` and enter your Odoo database URL followed by

Reviewer 3: Missing step.
```suggestion
#. | Go to :menuselection:`Settings --> Webhooks`, click on :guilabel:`Create New Webhook`, and enter your Odoo database URL followed by
```

     `/payment/razorpay/webhook` in the :guilabel:`Webhook URL` text field.

>    | For example: `https://yourcompany.odoo.com/payment/razorpay/webhook`.

Reviewer 2: ```suggestion
   | For example: `https://example.odoo.com/payment/razorpay/webhook`.
```

#. Fill the :guilabel:`Secret` field with a password of your choice and save it for later.

> #. Make sure  :guilabel:`refund.processed` and :guilabel:`payment.captured` checkboxes are ticked.

Reviewer 3: Swap to follow Razorpay's order.
```suggestion
#. Make sure the :guilabel:`payment.captured` and :guilabel:`refund.processed` checkboxes are ticked.
```

> #. Click on **Save** to finalize the configuration.

Reviewer: This one slipped my attention, it should be :guilabel:`Save` since this a GUI element. My bad! :)

Author: @toaa-odoo , Done..! Thanks

> #. Click on :guilabel:`Save` to finalize the configuration.

Reviewer 3: ```suggestion
#. Click on :guilabel:`Create Webhook` to finalize the configuration.
```

.. _payment_acquirers/razorpay/configure_odoo:

Configuration on Odoo
=====================

#. :ref:`Navigate to the payment acquirer Razorpay <payment_acquirers/add_new>` and change its
   state to :guilabel:`Enabled`.
#. In the :guilabel:`Credentials` tab, fill the :guilabel:`Public Key`, :guilabel:`Secret Key`, and
   :guilabel:`Webhook Secret` with the values you saved at the step
   :ref:`payment_acquirers/razorpay/configure_dashboard`.
#. Configure the rest of the options to your liking.

.. important::

>   - If you configure Odoo to capture amounts manually, make sure to set the **Manual Capture**

Reviewer 2: last missing guilabel haha ;)

>     to **manual** on Razorpay. Otherwise, the transaction will be blocked in the authorized

Reviewer 2: us guilabel as well if that's how it's displayed on razorpay's website.

>     to **manual** on Razorpay. Otherwise, the transaction will be blocked in the authorized

Reviewer 2: avoid the use of future for normal expected behaviour.
Here you could use "would"

    state in Odoo. To do so, go to your Razorpay Dashboard and then to
    :menuselection:`Settings --> Configuration`.

>   - After **5 days**, if the transaction hasn't been captured yet, the customer has the right

Reviewer 2: - numbers up to ten must be written in full
- 
```suggestion
  - After **five days**, if the transaction hasn't been captured yet, the customer has the right
```
